extension/llm/export/builder.py

```diff
@@ -167,6 +176,59 @@ def capture_pre_autograd_graph(self) -> "LLMEdgeManager":
             )
         return self
 
+
+    def pt2e_calibrate(
+        self,
+        prepared_module,
+        calibration_tasks,
+        calibration_limit,
+        calibration_seq_length,
+        tokenizer_path,
+    ):
+        logging.info("Run calibration...")
+        try:
+            from executorch.examples.models.llama2.evaluate import EagerEvalWrapper, evaluate_model
+        except ImportError:
+            raise ImportError(
+                "Please install the llm eval dependency via examples/models/llama2/install_requirements.sh"
+            )
+
+        tokenizer = get_tokenizer(tokenizer_path)
+
+        def calibrate_template(module: torch.fx.GraphModule, tokenizer, string: str = "Once upon a time", max_len: int = 128):
+                # TODO: change criteria & support batch inputs if necessary
+                pos = torch.tensor(0, dtype=torch.int64)
+                token_list = [tokenizer.bos_id] + tokenizer.encode(string, bos=True, eos=False)
+
+                with torch.no_grad():
+                    while token_list[-1] != tokenizer.eos_id and pos < max_len:
```

Comment:
For llama 3, `eos_id` is actually `eot_id` BTW, https://huggingface.co/meta-llama/Meta-Llama-3-8B-Instruct/discussions/73

Reply:
Curious why not batch prefill here to make it faster?

Reply:
I think if batch prefill here, it should also check dynamic shape.

Reply:
Here we use the graph module instead of eager model for calibration, and the graph module is captured with static shape. The graph module is captured with fix shape and batch prefill requires dynamic shape.